### ISSUE TITLE
Unable to create a PEX with an unpublished wheel with a lock file

### ISSUE THREAD

**Issue by zmanji** [pex-tool/pex#1967]
I have an unpublished wheel with requirements. Think a native code wheel that has requirements in pyproject.toml. Given this wheel I want to build a pex but I want the resolve to use a lock for reproducibility. 

What I have done is create a lockfile with all of the third party deps I need.

When I try:
```
pex --lock pex.lock <WHL>
```

This fails with:
```
Failed to resolve compatible artifacts from lock pex.lock for 1 target:
1. /home/zmanji/.pex/venvs/ad3fcfb926d70cf748f6693d03b37390c95022dc/5985ed09b49a653d6596b0e14d134c5456cf1a9f/bin/python3.10:
    Failed to resolve all requirements for cp310-cp310-manylinux_2_35_x86_64 interpreter at /home/zmanji/.pex/venvs/ad3fcfb926d70cf748f6693d03b37390c95022dc/5985ed09b49a653d6596b0e14d134c5456cf1a9f/bin/python3.10 from pex.lock:

Configured with:
    build: True
    use_wheel: True

Dependency on <PROJECT> (via: <PROJECT>==0.1) not satisfied, no candidates found.
```

I have the same issue when I put all of the third party deps into a pex and ask pex to resolve from a pex repository.

I also have tried building an intransitive pex with just the wheel and putting all of the other deps in another pex that's put on the PEX_PATH but pex does not traverse the entire pex path when activating a pex. I think this was discussed in https://github.com/pantsbuild/pex/issues/1423 

Currently I resolve my issue by:
1. Building a pex with all third party dependencies.
2. Extracting all third party dependencies into a wheelhouse
3. Build a final pex with my projects wheel, with no pypi index and the above wheelhouse.

My workaround is fine so I don't think this is an urgent to be fixed but I think https://github.com/pantsbuild/pants/issues/15801 is the same issue in pants. 

I do think this is worth resolving because conceptually to me resolving from a pex-repository (or a lock file) should be the same as resolving from a fixed wheelhouse.

**Comment by jsirois** [comment]
@zmanji I think #2455 addresses this. That change adds `--project` to both `pex3 lock {create,sync}` and `pex`. Using that feature, instead of pre-building a wheel as you describe, you'd just:
1. `pex3 lock sync --project . --lock lock.json`
2. `pex --project . --lock lock.json`

Have I understood what you're trying to do here or do you actually really need to pre-build the wheel as you've described?

**Comment by jsirois** [comment]
@zmanji `--project` is now available in the 2.8.0 release if you still have the OP use case and want to try this out in that context.

**Comment by jsirois** [comment]
> I have an unpublished wheel with requirements. Think a native code wheel that has requirements in pyproject.toml.

@zmanji I'm going to close this based on the second quoted sentence. That problem is solved with `--project` where you let Pex (Pip really) build the wheel from your pyproject.toml. If, for some reason, you must build the wheel out of band (or otherwise obtain it out of band) then please re-open with a comment explaining that use case a bit more.
